### PR TITLE
[track_10min_16mb] utkuaydiin 10L Muon — 15.46 MB

### PR DESCRIPTION
**Pull request by utkuaydiin**
Pure modeling track submission.

- 10 layers, 448 dim, BigramHash(8192) + SmearGate
- Muon optimizer + SWA + 5.5% global prune + mixed Int5/Int6 PTQ
- Trained 10 min on 8×H100 with real FineWeb data
- Final artifact: 15.46 MB after zstd-22

### CODE REVIEW

File: records/track_10min_16mb/2026-03-23_utkuaydiin_10L_Muon/train_gpt.py [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+    if not os.path.exists(filename):
+        print("WARNING: Using random tokens")
+        x = torch.randint(0, 1024, (batch_size, block_size))
+        y = torch.randint(0, 1024, (batch_size, block_size))
+        return x.to(device), y.to(device)
+
+    data = np.memmap(filename, dtype=np.uint16, mode='r')
+    ix = torch.randint(len(data) - block_size, (batch_size,))
+    x = torch.stack([torch.from_numpy((data[i:i+block_size]).astype(np.int64)) for i in ix])
+    y = torch.stack([torch.from_numpy((data[i+1:i+1+block_size]).astype(np.int64)) for i in ix])
```
[the remark this answers is not in the document]
Comment (utkuaydiin) [reply records/track_10min_16mb/2026-03-23_utkuaydiin_10L_Muon/train_gpt.py]: @copilot if len(data) <= block_size + 1:
        # Instead of random tokens, we fail fast as requested by maintainers
        raise ValueError(f"FineWeb shard {filename} is too small or malformed (Tokens: {len(data)})")
